revocation.md

```diff
@@ -0,0 +1,237 @@
+# Revocation
+
+**Patent Pending**
+
+Revocation requires that a verifiable revocation notice for a given Doughnut be posted somewhere
+accessible to all Doughnut revokers and verifiers.
+
+Because Doughnuts support decentralisation it makes sense for a Doughnut to tell verifiers
+how and where to revoke it, and where to check its revocation status.
+This removes the need for centralised revocation services.
+
+What follows are specifications for decentralised Doughnut revocation methods.
+A Doughnut version may specify a specific revocation method, or allow the issuer to
+make a choice of revocation method.
+
+Different strategies for achieving revocation are explored in the [`Strategies`](#strategies) section below.
+
+
+
+
+
+# Methods
+
+
+
+## 0
```

Comment:
I would suggest we support revocation through non HTTP - distributed services such as a smart contract address on a specified blockchain network. This means in the doughnut we may need a field that would say what kind of revocation service is desired. Something like:
`revocation: Number`
Then if it's 0, the revocation is through an HTTP server, if 1 by a smart contract. Then if it's a smart contract, instead of revocation_uri, we'd have network type (ETH, CENNZNet, ..), Network name (Azlea) and contract address(0x...). We may also say in the case of a smart contract, the revocation_uri should be composed of the aforementioned fields.